### DISCUSSION TITLE
关于form.values的数据结构问题

### DISCUSSION THREAD

**Discussion by zhouxinyong** [alibaba/formily#2497] [Q&A 2.x] [answered]
codesandbox地址：https://codesandbox.io/s/formily-bug-iztlq?file=/App.tsx

初始化后点击提交
form.values 为 
![image](https://user-images.githubusercontent.com/15569753/143191389-6c56ff5f-3f24-4a58-a8f4-c3ff5131d5e0.png)

然后点击季度，再点击提交
form.values变成了
![image](https://user-images.githubusercontent.com/15569753/143191836-b68b8ca3-f7c5-44da-8946-11aa726f9c76.png)

这个是bug还是我的配置问题？

**Comment by janryWang** [comment, marked as answer]
自定义组件内部会对日期格式做强制转换

**Reply by zhouxinyong** [reply]
下拉框的数据也没了

**Reply by janryWang** [reply]
不要传Date数据，统一传YYYY-MM-DD这种

**Reply by janryWang** [reply]
哦，我知道了，因为你给Row组件传了type object，Antd的Radio.Group组件内部触发onChange会直接冒泡给Row，然后Row组件就会触发onChange了，建议再套一层object，布局组件上只用void
